src/components/toolbar/helpers.ts

```diff
@@ -228,37 +228,31 @@ export const promoteParagraphToSection = (
   }
 
   if (parentSection.type.name === 'body') {
-    const resolvePos = findChildrenByType(parentSection, nodes.section)[0]
-    if (resolvePos) {
-      const { pos } = resolvePos
-      items[0] = nodes.section.create(
-        {},
-        Fragment.from(sectionTitle).append(
-          parentSection.slice(afterParagraphOffset, pos).content
-        )
-      )
+    const firstSection = findChildrenByType(parentSection, nodes.section)[0]
+    let content = parentSection.slice(afterParagraphOffset).content
+    if (firstSection) {
+      const { pos } = firstSection
+      content = parentSection.slice(afterParagraphOffset, pos).content
       afterParentSection = beforeParentSection + pos
       beforeParentSection = beforeParagraph
-    } else {
-      items[0] = nodes.section.create(
-        {},
-        Fragment.from(sectionTitle).append(
-          parentSection.slice(afterParagraphOffset).content
-        )
-      )
     }
 
+    items[0] = nodes.section.create(
+      {},
+      Fragment.from(sectionTitle).append(content)
+    )
+
     if ($beforeParagraph.index() === 0) {
```

Comment:
Could you please give this value a descriptive name or add a comment explaining the case you're trying to cover? This will help improve readability.

Reply:
updated, and added a comment: that will be the remaining content without sections at body level